contributors/design-proposals/raw-block-pv.md

```diff
@@ -0,0 +1,568 @@
+
+               Milestone 1: API changes
+
+               Milestone 2: Restricted Access 
+               
+               Milestone 3: Scrubbing device after PV is deleted
+               
+               Milestone 4: Changes to the mounter interface as today it is assumed 'file' as the default.
+               
+               Milestone 5: Expose volumeType to users via kubectl
+               
+               Milestone 6: Adds enable/disable configuration to securityContext in PSP (Pod Security Policy) similar to hostPath
+               
+               Milestone 7: Validate container runtime options with user specifcations as indicated in UC3
+	       
+	       Milestone 8: Container Runtime changes
+
+Phase 2:  v1.9
+Feature: Discovery of block devices 
+
+                Milestone 1: Dynamically provisioned PVs to dynamically allocated devices
+
+                Milestone 2: Privileged container concerns    
+
+Other considerations:
+
+               Reference volume driver change(Attach/Detach logic) for pre-provisioned PVs
+
+               Reference volume driver change(Provision/Delete logic) for dynamic provisioning
+
+## Mounter interface proposed design
+
+```
+ type BlockMounter interface {
+	Mounter
+ 	CanSupportFileOnBlock() error
+            // kubelet needs volume's device path when attaches a volume to container
+ 	GetVolumePath() string
+ 	GetVolumeType() string   //TBD
+ }
+
+type BlockUnmounter interface {
+ 	Unmounter
+ 	GetVolumePath() string
+}
+```
+# Mounter binding matrix for statically provisioned volumes:
+
+| PV volumeType | PVC volumeType  | Result           |
+| --------------|:---------------:| ----------------:|
+|   unspecified | unspecified     | BIND             |
+|   file        | file            | BIND             |
+|   block       | unspecified     | NO BIND          |
+|   block       |  block          | BIND             |
+|   unspecified | block           | NO BIND          |
+|   block       |  file           | NO BIND          |
+|   file        |  block          | NO BIND          |
+|  unspecified  | file            | BIND             |
+
+
+* unspecified defaults to 'file/ext4' today for backwards compatibility and in mount_linux.go
+
+# Mounter binding matrix for dynamically provisioned volumes:
+
+Note: The value used for the plugin to indicate is it provisioning 
+block will be plugin dependent and is an opaque parameter. Thus, not an API
+change and possibly inconsistent between plugins. We are suggesting using 'block'
+to simplify validation in the code (rather than raw that what proposed before).
+
+| PV volumeType | Plugin fstype | PVC volumeType  | Result           |
+| --------------|:-------------:| ---------------:|-----------------:|
+|  --           | ext4/xfs      | block           | NO BIND          |
+|  --           | ext4/xfs      | unspecified     | BIND             |
+|  --           | block         | block           | BIND             |
```

Comment:
Today the fstype is not applied until the first mount so it plays no role in provisioning/binding.  I think we can remove it from the matrix to avoid confusion